### PR TITLE
add version info in ImportError's

### PR DESCRIPTION
**Pull request by raybellwaves**
This PR adds the version info in the `ImportError` message. I find this useful as in the past I have received the message `ImportError: Numba needs NumPy 1.22 or less` without immediately knowing with Numpy version I was using.

I apologize the code has been formatted by black: the default formatter in my text editor. If you require this to be removed  I'll be happy to do so.

The code edited is shown below with the new code (added `Got {VERSION}.` on the end of the some `ImportError` messages.

```
    if PYVERSION < (3, 7):
        raise ImportError(f"Numba needs Python 3.7 or greater. Got {PYVERSION}.")

    if numpy_version < (1, 18):
        raise ImportError(f"Numba needs NumPy 1.18 or greater. Got {numpy_version}.")

    try:
        import scipy
    except ImportError:
        pass
    else:
        sp_version = tuple(map(int, scipy.__version__.split(".")[:2]))
        if sp_version < (1, 0):
            raise ImportError(
                f"Numba requires SciPy version 1.0 or greater. Got {sp_version}."
            )
```

Thanks for your work on this package.

### CODE REVIEW

**Comment by esc** [comment]
@rawwar thank you for submitting this, it has now be labeled for review and hopefully we can assign a reviewer during the developer meeting tomorrow.

**Comment by stuartarchibald** [comment]
@raybellwaves thanks for implementing this, I think this is a good idea, it'll help in diagnosing issue with environments that are in an inconsistent state as well as those which are unsuitable.

RE:

> I apologize the code has been formatted by black: the default formatter in my text editor. If you require this to be removed I'll be happy to do so.

Yes please, if you could remove them it would be much appreciated, Numba tries to keep automated changes to a minimum to help with reducing conflicts etc. Once that's done I'll give this a review, thanks for your help!!

File: numba/__init__.py [hunk 1]
```diff
@@ -219,4 +221,4 @@ def _try_enable_svml():
 # a bind to the underlying LLVM libraries which then irretrievably sets the LLVM
 # SVML state to "no SVML". See https://github.com/numba/numba/issues/4689 for
 # context.
-# ---------------------- WARNING WARNING WARNING ----------------------------
+# ---------------------- WARNING WARNING WARNING ----------------------------
```
Comment (stuartarchibald) [inline numba/__init__.py:224]: Please could a newline be added back in here, I think it was accidentally removed? Thanks.

File: numba/__init__.py [hunk 2]
```diff
@@ -138,10 +138,10 @@ def _ensure_critical_deps():
     from numba.core.utils import PYVERSION
 
     if PYVERSION < (3, 7):
-        raise ImportError("Numba needs Python 3.7 or greater")
+        raise ImportError(f"Numba needs Python 3.7 or greater. Got {PYVERSION}.")
```
Comment (stuartarchibald) [inline numba/__init__.py:141]:
```suggestion
        msg = ("Numba needs Python 3.7 or greater. Got Python "
               f"{PYVERSION[0]}.{PYVERSION[1]}.")
        raise ImportError(msg)
```

File: numba/__init__.py [hunk 3]
```diff
@@ -150,7 +150,9 @@ def _ensure_critical_deps():
     else:
         sp_version = tuple(map(int, scipy.__version__.split('.')[:2]))
         if sp_version < (1, 0):
-            raise ImportError("Numba requires SciPy version 1.0 or greater")
+            raise ImportError(
+                f"Numba requires SciPy version 1.0 or greater. Got {sp_version}."
+            )
```
Comment (stuartarchibald) [inline numba/__init__.py:155]:
```suggestion
            msg = ("Numba requires SciPy version 1.0 or greater. Got SciPy "
                   f"{scipy.__version__}.")
            raise ImportError(msg)
```

File: numba/__init__.py [hunk 4]
```diff
@@ -138,10 +138,10 @@ def _ensure_critical_deps():
     from numba.core.utils import PYVERSION
 
     if PYVERSION < (3, 7):
-        raise ImportError("Numba needs Python 3.7 or greater")
+        raise ImportError(f"Numba needs Python 3.7 or greater. Got {PYVERSION}.")
 
     if numpy_version < (1, 18):
-        raise ImportError("Numba needs NumPy 1.18 or greater")
+        raise ImportError(f"Numba needs NumPy 1.18 or greater. Got {numpy_version}.")
```
Comment (stuartarchibald) [inline numba/__init__.py:144]:
```suggestion
        msg = (f"Numba needs NumPy 1.18 or greater. Got NumPy "
               f"{numpy_version[0]}.{numpy_version[1]}.")
        raise ImportError(msg)
```

**Review by stuartarchibald — COMMENTED** [review]
@raybellwaves thanks for the updates to the patch.

I tested this patch manually by setting the appropriate variables to unsupported version numbers and then trying to import. One observation was that the current error messages are of the form:
```
ImportError: Numba needs Python 3.7 or greater. Got (3, 5).
```
i.e. the "Got" version is the raw version tuple. I've made some suggestions inline to change this to something more like:
```
ImportError: Numba needs Python 3.7 or greater. Got Python 3.5.
```
what do you think?

Thanks for your input!

**Review by stuartarchibald — APPROVED** [review]
Many thanks for the patch and updates @raybellwaves. I gave 62b04bb a manual test similar to that in my last review, it seems to work as expected. Thanks again!

**Pull request merged**